src/diversity/components.py

```diff
@@ -0,0 +1,91 @@
+from abc import ABC, abstractmethod
+from functools import cached_property
+from typing import Union
+
+from numpy import ndarray
+
+from diversity.abundance import Abundance
+from diversity.similarity import Similarity
+
+
+class Components(ABC):
+    """Dispatches diversity components based on specified measure"""
+
+    @abstractmethod
+    def get_numerator(measure: str):
+        pass
+
+    @abstractmethod
+    def get_denominator(measure: str):
+        pass
+
+
+class FrequencySensitiveComponents(Components):
```

Comment:
I have flip flopped back and forth so many times on this terminology. The problem is that SimilarityInsensitiveComponents also makes no reference to frequency. My reasoning for keeping frequency in the name is because frequency is what people associate with traditional diversity measures. I think we could just be explicit in the docs and readme that Leinster diversity is both similarity- and frequency-sensitive.